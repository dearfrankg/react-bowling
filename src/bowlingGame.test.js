import BowlingGame from "./bowlingGame";

describe("BowlingGame kata", () => {
  let game;

  beforeEach(function() {
    game = BowlingGame.create();
  });

  const rollMany = (n, pins) => {
    while (n--) {
      game.roll(pins);
    }
  };

  function rollSpare() {
    game.roll(5);
    game.roll(5);
  }

  function rollStrike() {
    game.roll(10);
  }

  it("should handle a gutter game", () => {
    rollMany(20, 0);
    expect(game.score()).toMatchSnapshot();
  });

  it("should handle all ones", () => {
    rollMany(20, 1);
    expect(game.score()).toMatchSnapshot();
  });

  it("should handle one spare", () => {
    rollSpare();
    game.roll(3);
    rollMany(17, 0);
    expect(game.score()).toMatchSnapshot();
  });

  it("should handle one strike", () => {
    rollStrike();
    game.roll(3);
    game.roll(4);
    rollMany(16, 0);
    expect(game.score()).toMatchSnapshot();
  });

  it("should handle a perfect game", () => {
    rollMany(12, 10);
    expect(game.score()).toMatchSnapshot();
  });
});
