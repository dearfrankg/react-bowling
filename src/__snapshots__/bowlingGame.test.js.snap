// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BowlingGame kata should handle a gutter game 1`] = `
Array [
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 0,
    "extraBox": "",
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
]
`;

exports[`BowlingGame kata should handle a perfect game 1`] = `
Array [
  Object {
    "cumulativeScore": 30,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 60,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 90,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 120,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 150,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 180,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 210,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 240,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 270,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 300,
    "extraBox": "X",
    "leftBox": "X",
    "pinsUp": 10,
    "rightBox": "X",
  },
]
`;

exports[`BowlingGame kata should handle all ones 1`] = `
Array [
  Object {
    "cumulativeScore": 2,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 4,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 6,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 8,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 10,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 12,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 14,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 18,
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
  Object {
    "cumulativeScore": 20,
    "extraBox": "",
    "leftBox": 1,
    "pinsUp": 10,
    "rightBox": 1,
  },
]
`;

exports[`BowlingGame kata should handle one spare 1`] = `
Array [
  Object {
    "cumulativeScore": 13,
    "leftBox": 5,
    "pinsUp": 10,
    "rightBox": "/",
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 3,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 16,
    "extraBox": "",
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
]
`;

exports[`BowlingGame kata should handle one strike 1`] = `
Array [
  Object {
    "cumulativeScore": 17,
    "leftBox": "",
    "pinsUp": 10,
    "rightBox": "X",
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 3,
    "pinsUp": 10,
    "rightBox": 4,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
  Object {
    "cumulativeScore": 24,
    "extraBox": "",
    "leftBox": 0,
    "pinsUp": 10,
    "rightBox": 0,
  },
]
`;
